sdk/data/azappconfig/client.go

```diff
@@ -60,24 +61,25 @@ func newClient(endpoint string, authPolicy policy.Policy, options *ClientOptions
 		options = &ClientOptions{}
 	}
 
-	syncTokenPolicy := newSyncTokenPolicy()
-
+	cache := synctoken.NewCache()
 	client, err := azcore.NewClient(moduleName+".Client", moduleVersion, runtime.PipelineOptions{
-		PerRetry: []policy.Policy{authPolicy, syncTokenPolicy},
+		PerRetry: []policy.Policy{authPolicy, synctoken.NewPolicy(cache)},
 	}, &options.ClientOptions)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Client{
 		appConfigClient: generated.NewAzureAppConfigurationClient(endpoint, client),
-		syncTokenPolicy: syncTokenPolicy,
+		cache:           cache,
 	}, nil
 }
 
-// UpdateSyncToken sets an external synchronization token to ensure service requests receive up-to-date values.
-func (c *Client) UpdateSyncToken(token string) {
-	c.syncTokenPolicy.addToken(token)
+// SetSyncToken is used to set a sync token from an external source.
+// Sync tokens are required to be in the format "<id>=<value>;sn=<sn>".
+// See https://learn.microsoft.com/en-us/azure/azure-app-configuration/rest-api-consistency for more information on sync tokens.
+func (c *Client) SetSyncToken(syncToken string) error {
```

Comment:
While this does make us deviate from the other SDKs, they got it wrong.

CC @JeffreyRichter

Reply:
OK. Do we want to introduce a "type SyncToken string" with some helper methods for better type-safety, splitting, and maybe determining with version is later?

Reply:
We could add a `type SyncToken string` and replace all instances of `SyncToken *string` with it.  However, I don't think any of the splitting methods need to be public as only the `synctoken.Cache` cares about that (I don't see customers ever needing the ability to perform these operations; plus, we could always make them public later).

As a follow-up, how do you feel about making `SyncToken` fields _not_ pointer-to-type?  According to the docs, _all_ operations return a Sync-Token value so it would never be `nil`.  Plus, Sync-Token can never be an empty string, so I think an empty check instead of a `nil` check is not ambiguous.

Reply:
Fine with me. The SyncToken type-safety improves the API. The methods could be added later (if we ever need them) and removing pointer is also OK with me if we're 100% sure that there is never a chance of nil today or in the future.

Reply:
Done.